Common/Data/Consolidators/RenkoConsolidator.cs

```diff
@@ -122,205 +105,63 @@ public RenkoConsolidator(decimal barSize, Func<IBaseData, decimal> selector, Fun
                 }
             }
             else
             {
-                _selector = (x => x.Value);
+                _selector = x => x.Value;
             }
 
             if (volumeSelector != null)
             {
                 if (!volumeSelector.TryConvertToDelegate(out _volumeSelector))
                 {
-                    throw new ArgumentException("Unable to convert parameter 'volumeSelector' to delegate type Func<IBaseData, decimal>");
+                    throw new ArgumentException(
+                        "Unable to convert parameter 'volumeSelector' to delegate type Func<IBaseData, decimal>");
                 }
             }
             else
             {
-                _volumeSelector = (x => 0);
+                _volumeSelector = x => 0;
             }
-        }
-
-        /// <summary>
-        /// Gets the kind of the bar
-        /// </summary>
-        public RenkoType Type { get; private set; }
 
-        /// <summary>
-        /// Gets the bar size used by this consolidator
-        /// </summary>
-        public decimal BarSize => _barSize;
-
-        /// <summary>
-        /// Gets the most recently consolidated piece of data. This will be null if this consolidator
-        /// has not produced any data yet.
-        /// </summary>
-        public IBaseData Consolidated { get; private set; }
-
-        /// <summary>
-        /// Gets a clone of the data being currently consolidated
-        /// </summary>
-        public IBaseData WorkingData => _currentBar?.Clone();
+            _evenBars = evenBars;
+        }
 
         /// <summary>
-        /// Gets the type consumed by this consolidator
+        /// Event handler that fires when a new piece of data is produced
         /// </summary>
-        public Type InputType => typeof(IBaseData);
+        public event EventHandler<RenkoBar> DataConsolidated;
 
         /// <summary>
-        /// Gets <see cref="RenkoBar"/> which is the type emitted in the <see cref="IDataConsolidator.DataConsolidated"/> event.
+        /// Event handler that fires when a new piece of data is produced
         /// </summary>
-        public Type OutputType => typeof(RenkoBar);
-
-        // Used for unit tests
-        internal RenkoBar OpenRenkoBar => new RenkoBar(null, _openOn, _closeOn, _barSize, _openRate, _highRate, _lowRate, _closeRate);
-
-        private void Rising(IBaseData data)
-        {
-            decimal limit;
-
-            while (_closeRate > (limit = _openRate + _barSize))
-            {
-                var wicko = new RenkoBar(data.Symbol, _openOn, _closeOn, _barSize, _openRate, limit, _lowRate, limit);
-
-                _lastWicko = wicko;
-
-                OnDataConsolidated(wicko);
-
-                _openOn = _closeOn;
-                _openRate = limit;
-                _lowRate = limit;
-            }
-        }
-
-        private void Falling(IBaseData data)
+        event DataConsolidatedHandler IDataConsolidator.DataConsolidated
```

Comment:
nit - would be nice to move `IDataConsolidator.DataConsolidated` & `event EventHandler<RenkoBar> DataConsolidated` above the constructors